lib/src/excel.dart

```diff
@@ -1,36 +1,128 @@
+      await ExcelJS.run(allowInterop((context) {
+        // Grabs the current worksheet
+        final sheet = context.workbook.worksheets.getActiveWorksheet();
+
+        // Adds a master table
+        final endAddressInt = _startAddressInt + _tableHeader.length - 1;
+        final tableAddress = 'A1:${String.fromCharCode(endAddressInt)}1';
+        final table = context.workbook.tables.add(tableAddress, true);
+
+        // Adds and formats table header
+        table.getHeaderRowRange()
+          ..values = [_tableHeader]
+          ..format.font.name = _fontName
+          ..format.font.size = _fontSize
+          ..format.font.bold = true
+          ..format.horizontalAlignment = _horizontalAlignment
+          ..format.borders.getItem('EdgeTop').style = _borderStyle
+          ..format.borders.getItem('EdgeBottom').style = _borderStyle;
+
+        // Adds and formats table body
+        table.getDataBodyRange()
+          ..formulas = _generateTableBody(insertionOrderList)
+          ..format.font.name = _fontName
+          ..format.font.size = _fontSize
+          ..format.horizontalAlignment = _horizontalAlignment;
+
+        // Auto-fits all used cells
+        sheet.getUsedRange().getEntireColumn().format.autofitColumns();
+        sheet.getUsedRange().getEntireRow().format.autofitRows();
+
+        // Sets the sheet as active
+        sheet.activate();
+        return context.sync();
+      }));
     }));
   }
 
-  Future<void> _populate(RequestContext context) async {
-    final sheet = context.workbook.worksheets.getActiveWorksheet();
-    final range = sheet.getRange('A1');
-    final values = <List<String>>[
-      ['test test']
-    ];
-    range.values = values;
-    range.format.autofitColumns();
-    return context.sync();
+  static List _generateTableBody(List insertionOrderList) {
+    final tableBody = [];
+
+    for (InsertionOrder io in insertionOrderList) {
+      tableBody.add([
+        io.insertionOrderId,
+        io.advertiserId,
+        io.campaignId,
+        io.displayName,
+        io.entityStatus.toString(),
+        io.updateTime,
+        io.pacing.pacingPeriod.toString(),
+        io.pacing.pacingType.toString(),
+        io.pacing.dailyMaxMicros,
+        io.pacing.dailyMaxImpressions,
+        io.budget.budgetUnit.toString(),
+        io.budget.automationType.toString(),
+        _calculateTotalBudgetAmount(io.budget.budgetSegments),
+        _getStartDate(io.budget.budgetSegments.first),
+        _getEndDate(io.budget.budgetSegments.last),
+      ]);
+    }
+    return tableBody;
+  }
+
+  static String _calculateTotalBudgetAmount(
+      List<InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment>
+          budgetSegments) {
+    var totalBudgetMicros = Int64.ZERO;
+
+    for (var segment in budgetSegments) {
+      totalBudgetMicros += Int64.parseInt(segment.budgetAmountMicros);
+    }
+
+    return (totalBudgetMicros.toDouble() / 1000000).toString();
```

Comment:
is the toDouble necessary for division? that could result in some precision loss
consider storing the 1000000 as a constant. or having a conversion function
also does this format it like we would expect money to be formatted? ie $##.## ?

Reply:
Also currency vs impression budgets?

Reply:
Also impression vs currency budget.

Reply:
The formatting will be done from the Excel side using their javascript APIs. Conditional format the budget amount column based on the budgetType (currency or impression) will be done in a separate PR. New issue created #34.

Reply:
i think we want to handle currency budget case end to end first then impression budget?